Source/WebCore/platform/graphics/BitmapImageSource.cpp

```diff
@@ -526,8 +526,10 @@ DecodingStatus BitmapImageSource::requestNativeImageAtIndexIfNeeded(unsigned ind
     if (index >= m_frames.size())
         return DecodingStatus::Invalid;
 
+    // Never decode the same frame from two different threads.
```

Comment:
We need a radar to reference here.

Reply:
I added a FIXME comment to remove this completely. ImageIO is supposed to handle this case safely and be thread safe when decoding the same frame from multiple threads.